### PR TITLE
changes

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: Model/ResourceModel/Campaign.php [hunk 1]
```diff
@@ -5,10 +5,84 @@
...
+     * Set error message
+     *
+     * @param $campaignId
+     * @param $message
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function setMessage($campaignId, $message)
+    {
+        try {
+            $now = $this->_dateTime->formatDate(true);
```
Comment (cdiacon) [inline Model/ResourceModel/Campaign.php]: @adeelq how about to remove the contructor and do? : 

$now = time();

Comment (adeelq) [inline Model/ResourceModel/Campaign.php]: Yup we can

File: Block/Adminhtml/Campaign/Grid.php [hunk 2]
```diff
@@ -94,7 +94,16 @@ protected function _prepareColumns()
                 'null' => 'Not Send',
             ],
             'filter_condition_callback' => [$this, 'filterCallbackContact'],
-        ])->addColumn('message', [
+        ])->addColumn(
+            'send_id', array(
```
Comment (cdiacon) [inline Block/Adminhtml/Campaign/Grid.php]: @adeelq use this [] instead array()

File: Setup/InstallSchema.php [hunk 3]
```diff
@@ -418,6 +418,27 @@ public function install(
         );
         $installer->getConnection()->query($sqlQuery);
 
+        //Update contacts with customers that are subscribers
+        $select = $installer->getConnection()->select();
+
+        //join
+        $select->joinLeft(
+            array('ns' => $installer->getTable('newsletter_subscriber')),
```
Comment (cdiacon) [inline Setup/InstallSchema.php]: @adeelq use this [] instead array()

File: Setup/UpgradeSchema.php [hunk 4]
```diff
@@ -22,6 +22,20 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             //remove quote table
             $connection->dropTable($setup->getTable('email_quote'));
         }
+        if (version_compare($context->getVersion(), '2.0.4') < 0) {
+            //modify email_campaign table
+            $campaignTable = $setup->getTable('email_campaign');
+
+            //add column
+            $connection->addColumn(
+                $campaignTable, 'send_id', array(
```
Comment (cdiacon) [inline Setup/UpgradeSchema.php]: @adeelq use this [] instead array()

File: Model/Sync/Automation.php [hunk 5]
```diff
@@ -167,22 +167,26 @@ public function sync()
...
+                        $email, $this->websiteId
                     );
-                    $contacts[$automation->getWebsiteId()]['contacts'][$automation->getId()] = $contactId;
-                } else {
-                    // the contact is suppressed or the request failed
-                    //@codingStandardsIgnoreStart
-                    $automation->setEnrolmentStatus('Suppressed')
-                        ->save();
-                    //@codingStandardsIgnoreEnd
+                    //contact id is valid, can update datafields
```
Comment (cdiacon) [inline Model/Sync/Automation.php]: @adeelq 2 more 
line 18
line 154

**Pull request merged**